identity/src/commonMain/kotlin/com/android/identity/securearea/cloud/CloudUserAuthType.kt

```diff
@@ -0,0 +1,43 @@
+package com.android.identity.securearea.cloud
+
+/**
+ * An enumeration for different user authentication types when using [CloudSecureArea].
+ */
+enum class CloudUserAuthType(val flagValue: Long) {
+    /**
+     * Flag indicating that authentication is needed using the user's knowledge
+     * factor for Device Lock, e.g. passcode on iOS or LSKF on Android.
+     */
+    PASSCODE(1 shl 0),
+
+    /**
+     * Flag indicating that authentication is needed using the user's biometric.
+     */
+    BIOMETRIC(1 shl 1);
+
+    companion object {
+        /**
+         * Helper to encode a set of [CloudUserAuthType] as an integer.
+         */
+        fun encodeSet(types: Set<CloudUserAuthType>): Long {
+            var value = 0L
+            for (type in types) {
+                value = value or type.flagValue
+            }
+            return value
+        }
+
+        /**
+         * Helper to decode an integer into a set of [CloudUserAuthType].
+         */
+        fun decodeSet(types: Long): Set<CloudUserAuthType> {
+            val result = mutableSetOf<CloudUserAuthType>()
```

Comment:
Do we want to check for stray bits that aren't one of the recognized flags?

Reply:
I'm not sure there's anything meaningful we can do except print a warning? Realistically, the onus will be on the app using these to ensure that the two ends are using the same version of the code... entirely likely they're not - by design - and that the desired outcome is that stray bits are ignored.

I've added to the docs that we are just ignoring bits not corresponding to authentication types.